{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Manual de la Nave"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: ipywidgets in c:\\users\\eleaquim\\appdata\\local\\programs\\python\\python310-32\\lib\\site-packages (7.6.5)Note: you may need to restart the kernel to use updated packages.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "WARNING: Ignoring invalid distribution -ip (c:\\users\\eleaquim\\appdata\\local\\programs\\python\\python310-32\\lib\\site-packages)\n",
      "WARNING: Ignoring invalid distribution -ip (c:\\users\\eleaquim\\appdata\\local\\programs\\python\\python310-32\\lib\\site-packages)\n",
      "WARNING: Ignoring invalid distribution -ip (c:\\users\\eleaquim\\appdata\\local\\programs\\python\\python310-32\\lib\\site-packages)\n",
      "WARNING: Ignoring invalid distribution -ip (c:\\users\\eleaquim\\appdata\\local\\programs\\python\\python310-32\\lib\\site-packages)\n",
      "WARNING: Ignoring invalid distribution -ip (c:\\users\\eleaquim\\appdata\\local\\programs\\python\\python310-32\\lib\\site-packages)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Requirement already satisfied: ipython-genutils~=0.2.0 in c:\\users\\eleaquim\\appdata\\local\\programs\\python\\python310-32\\lib\\site-packages (from ipywidgets) (0.2.0)\n",
      "Requirement already satisfied: nbformat>=4.2.0 in c:\\users\\eleaquim\\appdata\\local\\programs\\python\\python310-32\\lib\\site-packages (from ipywidgets) (5.1.3)\n",
      "Requirement already satisfied: ipykernel>=4.5.1 in c:\\users\\eleaquim\\appdata\\roaming\\python\\python310-32\\site-packages (from ipywidgets) (6.9.0)\n",
      "Requirement already satisfied: ipython>=4.0.0 in c:\\users\\eleaquim\\appdata\\roaming\\python\\python310-32\\site-packages (from ipywidgets) (8.0.1)\n",
      "Requirement already satisfied: traitlets>=4.3.1 in c:\\users\\eleaquim\\appdata\\roaming\\python\\python310-32\\site-packages (from ipywidgets) (5.1.1)\n",
      "Requirement already satisfied: widgetsnbextension~=3.5.0 in c:\\users\\eleaquim\\appdata\\local\\programs\\python\\python310-32\\lib\\site-packages (from ipywidgets) (3.5.2)\n",
      "Requirement already satisfied: jupyterlab-widgets>=1.0.0 in c:\\users\\eleaquim\\appdata\\local\\programs\\python\\python310-32\\lib\\site-packages (from ipywidgets) (1.0.2)\n",
      "Requirement already satisfied: jupyter-client<8.0 in c:\\users\\eleaquim\\appdata\\roaming\\python\\python310-32\\site-packages (from ipykernel>=4.5.1->ipywidgets) (7.1.2)\n",
      "Requirement already satisfied: matplotlib-inline<0.2.0,>=0.1.0 in c:\\users\\eleaquim\\appdata\\roaming\\python\\python310-32\\site-packages (from ipykernel>=4.5.1->ipywidgets) (0.1.3)\n",
      "Requirement already satisfied: tornado<7.0,>=4.2 in c:\\users\\eleaquim\\appdata\\roaming\\python\\python310-32\\site-packages (from ipykernel>=4.5.1->ipywidgets) (6.1)\n",
      "Requirement already satisfied: debugpy<2.0,>=1.0.0 in c:\\users\\eleaquim\\appdata\\roaming\\python\\python310-32\\site-packages (from ipykernel>=4.5.1->ipywidgets) (1.5.1)\n",
      "Requirement already satisfied: nest-asyncio in c:\\users\\eleaquim\\appdata\\roaming\\python\\python310-32\\site-packages (from ipykernel>=4.5.1->ipywidgets) (1.5.4)\n",
      "Requirement already satisfied: jedi>=0.16 in c:\\users\\eleaquim\\appdata\\roaming\\python\\python310-32\\site-packages (from ipython>=4.0.0->ipywidgets) (0.18.1)\n",
      "Requirement already satisfied: black in c:\\users\\eleaquim\\appdata\\roaming\\python\\python310-32\\site-packages (from ipython>=4.0.0->ipywidgets) (22.1.0)\n",
      "Requirement already satisfied: backcall in c:\\users\\eleaquim\\appdata\\roaming\\python\\python310-32\\site-packages (from ipython>=4.0.0->ipywidgets) (0.2.0)\n",
      "Requirement already satisfied: prompt-toolkit!=3.0.0,!=3.0.1,<3.1.0,>=2.0.0 in c:\\users\\eleaquim\\appdata\\roaming\\python\\python310-32\\site-packages (from ipython>=4.0.0->ipywidgets) (3.0.28)\n",
      "Requirement already satisfied: stack-data in c:\\users\\eleaquim\\appdata\\roaming\\python\\python310-32\\site-packages (from ipython>=4.0.0->ipywidgets) (0.1.4)\n",
      "Requirement already satisfied: decorator in c:\\users\\eleaquim\\appdata\\roaming\\python\\python310-32\\site-packages (from ipython>=4.0.0->ipywidgets) (5.1.1)\n",
      "Requirement already satisfied: pygments in c:\\users\\eleaquim\\appdata\\roaming\\python\\python310-32\\site-packages (from ipython>=4.0.0->ipywidgets) (2.11.2)\n",
      "Requirement already satisfied: pickleshare in c:\\users\\eleaquim\\appdata\\roaming\\python\\python310-32\\site-packages (from ipython>=4.0.0->ipywidgets) (0.7.5)\n",
      "Requirement already satisfied: setuptools>=18.5 in c:\\users\\eleaquim\\appdata\\local\\programs\\python\\python310-32\\lib\\site-packages (from ipython>=4.0.0->ipywidgets) (58.1.0)\n",
      "Requirement already satisfied: colorama in c:\\users\\eleaquim\\appdata\\roaming\\python\\python310-32\\site-packages (from ipython>=4.0.0->ipywidgets) (0.4.4)\n",
      "Requirement already satisfied: jupyter-core in c:\\users\\eleaquim\\appdata\\roaming\\python\\python310-32\\site-packages (from nbformat>=4.2.0->ipywidgets) (4.9.1)\n",
      "Requirement already satisfied: jsonschema!=2.5.0,>=2.4 in c:\\users\\eleaquim\\appdata\\local\\programs\\python\\python310-32\\lib\\site-packages (from nbformat>=4.2.0->ipywidgets) (4.4.0)\n",
      "Requirement already satisfied: notebook>=4.4.1 in c:\\users\\eleaquim\\appdata\\local\\programs\\python\\python310-32\\lib\\site-packages (from widgetsnbextension~=3.5.0->ipywidgets) (6.4.8)\n",
      "Requirement already satisfied: parso<0.9.0,>=0.8.0 in c:\\users\\eleaquim\\appdata\\roaming\\python\\python310-32\\site-packages (from jedi>=0.16->ipython>=4.0.0->ipywidgets) (0.8.3)\n",
      "Requirement already satisfied: pyrsistent!=0.17.0,!=0.17.1,!=0.17.2,>=0.14.0 in c:\\users\\eleaquim\\appdata\\local\\programs\\python\\python310-32\\lib\\site-packages (from jsonschema!=2.5.0,>=2.4->nbformat>=4.2.0->ipywidgets) (0.18.1)\n",
      "Requirement already satisfied: attrs>=17.4.0 in c:\\users\\eleaquim\\appdata\\local\\programs\\python\\python310-32\\lib\\site-packages (from jsonschema!=2.5.0,>=2.4->nbformat>=4.2.0->ipywidgets) (21.4.0)\n",
      "Requirement already satisfied: pyzmq>=13 in c:\\users\\eleaquim\\appdata\\roaming\\python\\python310-32\\site-packages (from jupyter-client<8.0->ipykernel>=4.5.1->ipywidgets) (22.3.0)\n",
      "Requirement already satisfied: python-dateutil>=2.1 in c:\\users\\eleaquim\\appdata\\roaming\\python\\python310-32\\site-packages (from jupyter-client<8.0->ipykernel>=4.5.1->ipywidgets) (2.8.2)\n",
      "Requirement already satisfied: entrypoints in c:\\users\\eleaquim\\appdata\\roaming\\python\\python310-32\\site-packages (from jupyter-client<8.0->ipykernel>=4.5.1->ipywidgets) (0.4)\n",
      "Requirement already satisfied: pywin32>=1.0 in c:\\users\\eleaquim\\appdata\\roaming\\python\\python310-32\\site-packages (from jupyter-core->nbformat>=4.2.0->ipywidgets) (303)\n",
      "Requirement already satisfied: nbconvert in c:\\users\\eleaquim\\appdata\\local\\programs\\python\\python310-32\\lib\\site-packages (from notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets) (6.4.2)\n",
      "Requirement already satisfied: Send2Trash>=1.8.0 in c:\\users\\eleaquim\\appdata\\local\\programs\\python\\python310-32\\lib\\site-packages (from notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets) (1.8.0)\n",
      "Requirement already satisfied: argon2-cffi in c:\\users\\eleaquim\\appdata\\local\\programs\\python\\python310-32\\lib\\site-packages (from notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets) (21.3.0)\n",
      "Requirement already satisfied: terminado>=0.8.3 in c:\\users\\eleaquim\\appdata\\local\\programs\\python\\python310-32\\lib\\site-packages (from notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets) (0.9.5)\n",
      "Requirement already satisfied: jinja2 in c:\\users\\eleaquim\\appdata\\local\\programs\\python\\python310-32\\lib\\site-packages (from notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets) (3.0.3)\n",
      "Requirement already satisfied: prometheus-client in c:\\users\\eleaquim\\appdata\\local\\programs\\python\\python310-32\\lib\\site-packages (from notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets) (0.13.1)\n",
      "Requirement already satisfied: wcwidth in c:\\users\\eleaquim\\appdata\\roaming\\python\\python310-32\\site-packages (from prompt-toolkit!=3.0.0,!=3.0.1,<3.1.0,>=2.0.0->ipython>=4.0.0->ipywidgets) (0.2.5)\n",
      "Requirement already satisfied: pathspec>=0.9.0 in c:\\users\\eleaquim\\appdata\\roaming\\python\\python310-32\\site-packages (from black->ipython>=4.0.0->ipywidgets) (0.9.0)\n",
      "Requirement already satisfied: tomli>=1.1.0 in c:\\users\\eleaquim\\appdata\\roaming\\python\\python310-32\\site-packages (from black->ipython>=4.0.0->ipywidgets) (2.0.1)\n",
      "Requirement already satisfied: mypy-extensions>=0.4.3 in c:\\users\\eleaquim\\appdata\\roaming\\python\\python310-32\\site-packages (from black->ipython>=4.0.0->ipywidgets) (0.4.3)\n",
      "Requirement already satisfied: click>=8.0.0 in c:\\users\\eleaquim\\appdata\\roaming\\python\\python310-32\\site-packages (from black->ipython>=4.0.0->ipywidgets) (8.0.3)\n",
      "Requirement already satisfied: platformdirs>=2 in c:\\users\\eleaquim\\appdata\\roaming\\python\\python310-32\\site-packages (from black->ipython>=4.0.0->ipywidgets) (2.5.0)\n",
      "Requirement already satisfied: asttokens in c:\\users\\eleaquim\\appdata\\roaming\\python\\python310-32\\site-packages (from stack-data->ipython>=4.0.0->ipywidgets) (2.0.5)\n",
      "Requirement already satisfied: pure-eval in c:\\users\\eleaquim\\appdata\\roaming\\python\\python310-32\\site-packages (from stack-data->ipython>=4.0.0->ipywidgets) (0.2.2)\n",
      "Requirement already satisfied: executing in c:\\users\\eleaquim\\appdata\\roaming\\python\\python310-32\\site-packages (from stack-data->ipython>=4.0.0->ipywidgets) (0.8.2)\n",
      "Requirement already satisfied: six>=1.5 in c:\\users\\eleaquim\\appdata\\roaming\\python\\python310-32\\site-packages (from python-dateutil>=2.1->jupyter-client<8.0->ipykernel>=4.5.1->ipywidgets) (1.16.0)\n",
      "Requirement already satisfied: pywinpty<1,>=0.5 in c:\\users\\eleaquim\\appdata\\local\\programs\\python\\python310-32\\lib\\site-packages (from terminado>=0.8.3->notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets) (0.5.7)\n",
      "Requirement already satisfied: argon2-cffi-bindings in c:\\users\\eleaquim\\appdata\\local\\programs\\python\\python310-32\\lib\\site-packages (from argon2-cffi->notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets) (21.2.0)\n",
      "Requirement already satisfied: MarkupSafe>=2.0 in c:\\users\\eleaquim\\appdata\\local\\programs\\python\\python310-32\\lib\\site-packages (from jinja2->notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets) (2.0.1)\n",
      "Requirement already satisfied: pandocfilters>=1.4.1 in c:\\users\\eleaquim\\appdata\\local\\programs\\python\\python310-32\\lib\\site-packages (from nbconvert->notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets) (1.5.0)\n",
      "Requirement already satisfied: mistune<2,>=0.8.1 in c:\\users\\eleaquim\\appdata\\local\\programs\\python\\python310-32\\lib\\site-packages (from nbconvert->notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets) (0.8.4)\n",
      "Requirement already satisfied: defusedxml in c:\\users\\eleaquim\\appdata\\local\\programs\\python\\python310-32\\lib\\site-packages (from nbconvert->notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets) (0.7.1)\n",
      "Requirement already satisfied: testpath in c:\\users\\eleaquim\\appdata\\local\\programs\\python\\python310-32\\lib\\site-packages (from nbconvert->notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets) (0.5.0)\n",
      "Requirement already satisfied: nbclient<0.6.0,>=0.5.0 in c:\\users\\eleaquim\\appdata\\local\\programs\\python\\python310-32\\lib\\site-packages (from nbconvert->notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets) (0.5.10)\n",
      "Requirement already satisfied: jupyterlab-pygments in c:\\users\\eleaquim\\appdata\\local\\programs\\python\\python310-32\\lib\\site-packages (from nbconvert->notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets) (0.1.2)\n",
      "Requirement already satisfied: bleach in c:\\users\\eleaquim\\appdata\\local\\programs\\python\\python310-32\\lib\\site-packages (from nbconvert->notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets) (4.1.0)\n",
      "Requirement already satisfied: cffi>=1.0.1 in c:\\users\\eleaquim\\appdata\\local\\programs\\python\\python310-32\\lib\\site-packages (from argon2-cffi-bindings->argon2-cffi->notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets) (1.15.0)\n",
      "Requirement already satisfied: webencodings in c:\\users\\eleaquim\\appdata\\local\\programs\\python\\python310-32\\lib\\site-packages (from bleach->nbconvert->notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets) (0.5.1)\n",
      "Requirement already satisfied: packaging in c:\\users\\eleaquim\\appdata\\local\\programs\\python\\python310-32\\lib\\site-packages (from bleach->nbconvert->notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets) (21.3)\n",
      "Requirement already satisfied: pycparser in c:\\users\\eleaquim\\appdata\\local\\programs\\python\\python310-32\\lib\\site-packages (from cffi>=1.0.1->argon2-cffi-bindings->argon2-cffi->notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets) (2.21)\n",
      "Requirement already satisfied: pyparsing!=3.0.5,>=2.0.2 in c:\\users\\eleaquim\\appdata\\local\\programs\\python\\python310-32\\lib\\site-packages (from packaging->bleach->nbconvert->notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets) (3.0.7)\n"
     ]
    }
   ],
   "source": [
    "pip install ipywidgets\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "69a442531ecc4511b5c9ed3281deed6b",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "ToggleButton(value=False, button_style='success', description='Iniciar Launch', icon='rocket', tooltip='Engage…"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "800dcf87592845e39c10dcb1b9ebd44a",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Output()"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import ipywidgets as widgets\n",
    "\n",
    "ignition = widgets.ToggleButton(\n",
    "    value=False,\n",
    "    description='Iniciar Launch',\n",
    "    button_style='success',\n",
    "    tooltip='Engage your Launch',\n",
    "    icon='rocket'\n",
    ")\n",
    "\n",
    "output = widgets.Output()\n",
    "\n",
    "display(ignition, output)\n",
    "\n",
    "def on_value_change(change):\n",
    "    with output:\n",
    "        if change['new'] == True:\n",
    "            print(\"Nave Iniciada!\")\n",
    "        else:   \n",
    "            print(\"Nave Detenida\")\n",
    "\n",
    "ignition.observe(on_value_change, names='value')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting matplotlibNote: you may need to restart the kernel to use updated packages.\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "WARNING: Ignoring invalid distribution -ip (c:\\users\\eleaquim\\appdata\\local\\programs\\python\\python310-32\\lib\\site-packages)\n",
      "WARNING: Ignoring invalid distribution -ip (c:\\users\\eleaquim\\appdata\\local\\programs\\python\\python310-32\\lib\\site-packages)\n",
      "  error: subprocess-exited-with-error\n",
      "  \n",
      "  × Building wheel for numpy (pyproject.toml) did not run successfully.\n",
      "  │ exit code: 1\n",
      "  ╰─> [201 lines of output]\n",
      "      Running from numpy source directory.\n",
      "      Cythonizing sources\n",
      "      Processing numpy/random\\_bounded_integers.pxd.in\n",
      "      Processing numpy/random\\bit_generator.pyx\n",
      "      C:\\Users\\Eleaquim\\AppData\\Local\\Temp\\pip-install-fiufllfs\\numpy_bbf9cef0562346869146f776e1fda8e4\\tools\\cythonize.py:63: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives\n",
      "        from distutils.version import LooseVersion\n",
      "      Processing numpy/random\\mtrand.pyx\n",
      "      Processing numpy/random\\_bounded_integers.pyx.in\n",
      "      Processing numpy/random\\_common.pyx\n",
      "      Processing numpy/random\\_generator.pyx\n",
      "      Processing numpy/random\\_mt19937.pyx\n",
      "      Processing numpy/random\\_pcg64.pyx\n",
      "      Processing numpy/random\\_philox.pyx\n",
      "      Processing numpy/random\\_sfc64.pyx\n",
      "      INFO: blas_opt_info:\n",
      "      INFO: blas_armpl_info:\n",
      "      INFO: No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      INFO: customize MSVCCompiler\n",
      "      INFO:   libraries armpl_lp64_mp not found in ['C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\lib', 'C:\\\\', 'C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\libs']\n",
      "      INFO:   NOT AVAILABLE\n",
      "      INFO:\n",
      "      INFO: blas_mkl_info:\n",
      "      INFO:   libraries mkl_rt not found in ['C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\lib', 'C:\\\\', 'C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\libs']\n",
      "      INFO:   NOT AVAILABLE\n",
      "      INFO:\n",
      "      INFO: blis_info:\n",
      "      INFO:   libraries blis not found in ['C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\lib', 'C:\\\\', 'C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\libs']\n",
      "      INFO:   NOT AVAILABLE\n",
      "      INFO:\n",
      "      INFO: openblas_info:\n",
      "      INFO:   libraries openblas not found in ['C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\lib', 'C:\\\\', 'C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\libs']\n",
      "      INFO: get_default_fcompiler: matching types: '['gnu', 'intelv', 'absoft', 'compaqv', 'intelev', 'gnu95', 'g95', 'intelvem', 'intelem', 'flang']'\n",
      "      INFO: customize GnuFCompiler\n",
      "      WARN: Could not locate executable g77\n",
      "      WARN: Could not locate executable f77\n",
      "      INFO: customize IntelVisualFCompiler\n",
      "      WARN: Could not locate executable ifort\n",
      "      WARN: Could not locate executable ifl\n",
      "      INFO: customize AbsoftFCompiler\n",
      "      WARN: Could not locate executable f90\n",
      "      INFO: customize CompaqVisualFCompiler\n",
      "      WARN: Could not locate executable DF\n",
      "      INFO: customize IntelItaniumVisualFCompiler\n",
      "      WARN: Could not locate executable efl\n",
      "      INFO: customize Gnu95FCompiler\n",
      "      WARN: Could not locate executable gfortran\n",
      "      WARN: Could not locate executable f95\n",
      "      INFO: customize G95FCompiler\n",
      "      WARN: Could not locate executable g95\n",
      "      INFO: customize IntelEM64VisualFCompiler\n",
      "      INFO: customize IntelEM64TFCompiler\n",
      "      WARN: Could not locate executable efort\n",
      "      WARN: Could not locate executable efc\n",
      "      INFO: customize PGroupFlangCompiler\n",
      "      WARN: Could not locate executable flang\n",
      "      WARN: don't know how to compile Fortran code on platform 'nt'\n",
      "      INFO:   NOT AVAILABLE\n",
      "      INFO:\n",
      "      INFO: accelerate_info:\n",
      "      INFO:   NOT AVAILABLE\n",
      "      INFO:\n",
      "      INFO: atlas_3_10_blas_threads_info:\n",
      "      INFO: Setting PTATLAS=ATLAS\n",
      "      INFO:   libraries tatlas not found in ['C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\lib', 'C:\\\\', 'C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\libs']\n",
      "      INFO:   NOT AVAILABLE\n",
      "      INFO:\n",
      "      INFO: atlas_3_10_blas_info:\n",
      "      INFO:   libraries satlas not found in ['C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\lib', 'C:\\\\', 'C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\libs']\n",
      "      INFO:   NOT AVAILABLE\n",
      "      INFO:\n",
      "      INFO: atlas_blas_threads_info:\n",
      "      INFO: Setting PTATLAS=ATLAS\n",
      "      INFO:   libraries ptf77blas,ptcblas,atlas not found in ['C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\lib', 'C:\\\\', 'C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\libs']\n",
      "      INFO:   NOT AVAILABLE\n",
      "      INFO:\n",
      "      INFO: atlas_blas_info:\n",
      "      INFO:   libraries f77blas,cblas,atlas not found in ['C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\lib', 'C:\\\\', 'C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\libs']\n",
      "      INFO:   NOT AVAILABLE\n",
      "      INFO:\n",
      "      C:\\Users\\Eleaquim\\AppData\\Local\\Temp\\pip-install-fiufllfs\\numpy_bbf9cef0562346869146f776e1fda8e4\\numpy\\distutils\\system_info.py:2077: UserWarning:\n",
      "          Optimized (vendor) Blas libraries are not found.\n",
      "          Falls back to netlib Blas library which has worse performance.\n",
      "          A better performance should be easily gained by switching\n",
      "          Blas library.\n",
      "        if self._calc_info(blas):\n",
      "      INFO: blas_info:\n",
      "      INFO:   libraries blas not found in ['C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\lib', 'C:\\\\', 'C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\libs']\n",
      "      INFO:   NOT AVAILABLE\n",
      "      INFO:\n",
      "      C:\\Users\\Eleaquim\\AppData\\Local\\Temp\\pip-install-fiufllfs\\numpy_bbf9cef0562346869146f776e1fda8e4\\numpy\\distutils\\system_info.py:2077: UserWarning:\n",
      "          Blas (http://www.netlib.org/blas/) libraries not found.\n",
      "          Directories to search for the libraries can be specified in the\n",
      "          numpy/distutils/site.cfg file (section [blas]) or by setting\n",
      "          the BLAS environment variable.\n",
      "        if self._calc_info(blas):\n",
      "      INFO: blas_src_info:\n",
      "      INFO:   NOT AVAILABLE\n",
      "      INFO:\n",
      "      C:\\Users\\Eleaquim\\AppData\\Local\\Temp\\pip-install-fiufllfs\\numpy_bbf9cef0562346869146f776e1fda8e4\\numpy\\distutils\\system_info.py:2077: UserWarning:\n",
      "          Blas (http://www.netlib.org/blas/) sources not found.\n",
      "          Directories to search for the sources can be specified in the\n",
      "          numpy/distutils/site.cfg file (section [blas_src]) or by setting\n",
      "          the BLAS_SRC environment variable.\n",
      "        if self._calc_info(blas):\n",
      "      INFO:   NOT AVAILABLE\n",
      "      INFO:\n",
      "      non-existing path in 'numpy\\\\distutils': 'site.cfg'\n",
      "      INFO: lapack_opt_info:\n",
      "      INFO: lapack_armpl_info:\n",
      "      INFO:   libraries armpl_lp64_mp not found in ['C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\lib', 'C:\\\\', 'C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\libs']\n",
      "      INFO:   NOT AVAILABLE\n",
      "      INFO:\n",
      "      INFO: lapack_mkl_info:\n",
      "      INFO:   libraries mkl_rt not found in ['C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\lib', 'C:\\\\', 'C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\libs']\n",
      "      INFO:   NOT AVAILABLE\n",
      "      INFO:\n",
      "      INFO: openblas_lapack_info:\n",
      "      INFO:   libraries openblas not found in ['C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\lib', 'C:\\\\', 'C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\libs']\n",
      "      INFO:   NOT AVAILABLE\n",
      "      INFO:\n",
      "      INFO: openblas_clapack_info:\n",
      "      INFO:   libraries openblas,lapack not found in ['C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\lib', 'C:\\\\', 'C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\libs']\n",
      "      INFO:   NOT AVAILABLE\n",
      "      INFO:\n",
      "      INFO: flame_info:\n",
      "      INFO:   libraries flame not found in ['C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\lib', 'C:\\\\', 'C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\libs']\n",
      "      INFO:   NOT AVAILABLE\n",
      "      INFO:\n",
      "      INFO: atlas_3_10_threads_info:\n",
      "      INFO: Setting PTATLAS=ATLAS\n",
      "      INFO:   libraries tatlas,tatlas not found in C:\\Users\\Eleaquim\\AppData\\Local\\Programs\\Python\\Python310-32\\lib\n",
      "      INFO:   libraries tatlas,tatlas not found in C:\\\n",
      "      INFO:   libraries tatlas,tatlas not found in C:\\Users\\Eleaquim\\AppData\\Local\\Programs\\Python\\Python310-32\\libs\n",
      "      INFO: <class 'numpy.distutils.system_info.atlas_3_10_threads_info'>\n",
      "      INFO:   NOT AVAILABLE\n",
      "      INFO:\n",
      "      INFO: atlas_3_10_info:\n",
      "      INFO:   libraries satlas,satlas not found in C:\\Users\\Eleaquim\\AppData\\Local\\Programs\\Python\\Python310-32\\lib\n",
      "      INFO:   libraries satlas,satlas not found in C:\\\n",
      "      INFO:   libraries satlas,satlas not found in C:\\Users\\Eleaquim\\AppData\\Local\\Programs\\Python\\Python310-32\\libs\n",
      "      INFO: <class 'numpy.distutils.system_info.atlas_3_10_info'>\n",
      "      INFO:   NOT AVAILABLE\n",
      "      INFO:\n",
      "      INFO: atlas_threads_info:\n",
      "      INFO: Setting PTATLAS=ATLAS\n",
      "      INFO:   libraries ptf77blas,ptcblas,atlas not found in C:\\Users\\Eleaquim\\AppData\\Local\\Programs\\Python\\Python310-32\\lib\n",
      "      INFO:   libraries ptf77blas,ptcblas,atlas not found in C:\\\n",
      "      INFO:   libraries ptf77blas,ptcblas,atlas not found in C:\\Users\\Eleaquim\\AppData\\Local\\Programs\\Python\\Python310-32\\libs\n",
      "      INFO: <class 'numpy.distutils.system_info.atlas_threads_info'>\n",
      "      INFO:   NOT AVAILABLE\n",
      "      INFO:\n",
      "      INFO: atlas_info:\n",
      "      INFO:   libraries f77blas,cblas,atlas not found in C:\\Users\\Eleaquim\\AppData\\Local\\Programs\\Python\\Python310-32\\lib\n",
      "      INFO:   libraries f77blas,cblas,atlas not found in C:\\\n",
      "      INFO:   libraries f77blas,cblas,atlas not found in C:\\Users\\Eleaquim\\AppData\\Local\\Programs\\Python\\Python310-32\\libs\n",
      "      INFO: <class 'numpy.distutils.system_info.atlas_info'>\n",
      "      INFO:   NOT AVAILABLE\n",
      "      INFO:\n",
      "      INFO: lapack_info:\n",
      "      INFO:   libraries lapack not found in ['C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\lib', 'C:\\\\', 'C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\libs']\n",
      "      INFO:   NOT AVAILABLE\n",
      "      INFO:\n",
      "      C:\\Users\\Eleaquim\\AppData\\Local\\Temp\\pip-install-fiufllfs\\numpy_bbf9cef0562346869146f776e1fda8e4\\numpy\\distutils\\system_info.py:1902: UserWarning:\n",
      "          Lapack (http://www.netlib.org/lapack/) libraries not found.\n",
      "          Directories to search for the libraries can be specified in the\n",
      "          numpy/distutils/site.cfg file (section [lapack]) or by setting\n",
      "          the LAPACK environment variable.\n",
      "        return getattr(self, '_calc_info_{}'.format(name))()\n",
      "      INFO: lapack_src_info:\n",
      "      INFO:   NOT AVAILABLE\n",
      "      INFO:\n",
      "      C:\\Users\\Eleaquim\\AppData\\Local\\Temp\\pip-install-fiufllfs\\numpy_bbf9cef0562346869146f776e1fda8e4\\numpy\\distutils\\system_info.py:1902: UserWarning:\n",
      "          Lapack (http://www.netlib.org/lapack/) sources not found.\n",
      "          Directories to search for the sources can be specified in the\n",
      "          numpy/distutils/site.cfg file (section [lapack_src]) or by setting\n",
      "          the LAPACK_SRC environment variable.\n",
      "        return getattr(self, '_calc_info_{}'.format(name))()\n",
      "      INFO:   NOT AVAILABLE\n",
      "      INFO:\n",
      "      INFO: numpy_linalg_lapack_lite:\n",
      "      INFO:   FOUND:\n",
      "      INFO:     language = c\n",
      "      INFO:\n",
      "      Warning: attempted relative import with no known parent package\n",
      "      C:\\Users\\Eleaquim\\AppData\\Local\\Programs\\Python\\Python310-32\\lib\\distutils\\dist.py:274: UserWarning: Unknown distribution option: 'define_macros'\n",
      "        warnings.warn(msg)\n",
      "      running bdist_wheel\n",
      "      running build\n",
      "      running config_cc\n",
      "      INFO: unifing config_cc, config, build_clib, build_ext, build commands --compiler options\n",
      "      running config_fc\n",
      "      INFO: unifing config_fc, config, build_clib, build_ext, build commands --fcompiler options\n",
      "      running build_src\n",
      "      INFO: build_src\n",
      "      INFO: building py_modules sources\n",
      "      creating build\n",
      "      creating build\\src.win32-3.10\n",
      "      creating build\\src.win32-3.10\\numpy\n",
      "      creating build\\src.win32-3.10\\numpy\\distutils\n",
      "      INFO: building library \"npymath\" sources\n",
      "      error: Microsoft Visual C++ 14.0 or greater is required. Get it with \"Microsoft C++ Build Tools\": https://visualstudio.microsoft.com/visual-cpp-build-tools/\n",
      "      [end of output]\n",
      "  \n",
      "  note: This error originates from a subprocess, and is likely not a problem with pip.\n",
      "  ERROR: Failed building wheel for numpy\n",
      "ERROR: Could not build wheels for numpy, which is required to install pyproject.toml-based projects\n",
      "WARNING: Ignoring invalid distribution -ip (c:\\users\\eleaquim\\appdata\\local\\programs\\python\\python310-32\\lib\\site-packages)\n",
      "WARNING: Ignoring invalid distribution -ip (c:\\users\\eleaquim\\appdata\\local\\programs\\python\\python310-32\\lib\\site-packages)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  Using cached matplotlib-3.5.1-cp310-cp310-win32.whl (7.1 MB)\n",
      "Collecting kiwisolver>=1.0.1\n",
      "  Using cached kiwisolver-1.3.2-cp310-cp310-win32.whl (42 kB)\n",
      "Requirement already satisfied: pyparsing>=2.2.1 in c:\\users\\eleaquim\\appdata\\local\\programs\\python\\python310-32\\lib\\site-packages (from matplotlib) (3.0.7)\n",
      "Collecting fonttools>=4.22.0\n",
      "  Using cached fonttools-4.29.1-py3-none-any.whl (895 kB)\n",
      "Collecting numpy>=1.17\n",
      "  Using cached numpy-1.22.2.zip (11.4 MB)\n",
      "  Installing build dependencies: started\n",
      "  Installing build dependencies: finished with status 'done'\n",
      "  Getting requirements to build wheel: started\n",
      "  Getting requirements to build wheel: finished with status 'done'\n",
      "  Preparing metadata (pyproject.toml): started\n",
      "  Preparing metadata (pyproject.toml): finished with status 'done'\n",
      "Requirement already satisfied: packaging>=20.0 in c:\\users\\eleaquim\\appdata\\local\\programs\\python\\python310-32\\lib\\site-packages (from matplotlib) (21.3)\n",
      "Requirement already satisfied: python-dateutil>=2.7 in c:\\users\\eleaquim\\appdata\\roaming\\python\\python310-32\\site-packages (from matplotlib) (2.8.2)\n",
      "Collecting pillow>=6.2.0\n",
      "  Using cached Pillow-9.0.1-cp310-cp310-win32.whl (2.8 MB)\n",
      "Collecting cycler>=0.10\n",
      "  Using cached cycler-0.11.0-py3-none-any.whl (6.4 kB)\n",
      "Requirement already satisfied: six>=1.5 in c:\\users\\eleaquim\\appdata\\roaming\\python\\python310-32\\site-packages (from python-dateutil>=2.7->matplotlib) (1.16.0)\n",
      "Building wheels for collected packages: numpy\n",
      "  Building wheel for numpy (pyproject.toml): started\n",
      "  Building wheel for numpy (pyproject.toml): finished with status 'error'\n",
      "Failed to build numpy\n",
      "Collecting numpy\n",
      "  Using cached numpy-1.22.2.zip (11.4 MB)\n",
      "  Installing build dependencies: started\n",
      "  Installing build dependencies: finished with status 'done'\n",
      "  Getting requirements to build wheel: started\n",
      "  Getting requirements to build wheel: finished with status 'done'\n",
      "  Preparing metadata (pyproject.toml): started\n",
      "  Preparing metadata (pyproject.toml): finished with status 'done'\n",
      "Building wheels for collected packages: numpy\n",
      "  Building wheel for numpy (pyproject.toml): started\n",
      "  Building wheel for numpy (pyproject.toml): finished with status 'error'\n",
      "Failed to build numpy\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "WARNING: Ignoring invalid distribution -ip (c:\\users\\eleaquim\\appdata\\local\\programs\\python\\python310-32\\lib\\site-packages)\n",
      "WARNING: Ignoring invalid distribution -ip (c:\\users\\eleaquim\\appdata\\local\\programs\\python\\python310-32\\lib\\site-packages)\n",
      "  error: subprocess-exited-with-error\n",
      "  \n",
      "  × Building wheel for numpy (pyproject.toml) did not run successfully.\n",
      "  │ exit code: 1\n",
      "  ╰─> [201 lines of output]\n",
      "      Running from numpy source directory.\n",
      "      Cythonizing sources\n",
      "      Processing numpy/random\\_bounded_integers.pxd.in\n",
      "      Processing numpy/random\\bit_generator.pyx\n",
      "      C:\\Users\\Eleaquim\\AppData\\Local\\Temp\\pip-install-_4vnv0i1\\numpy_267767f23bd142af96a61a29b987ffbb\\tools\\cythonize.py:63: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives\n",
      "        from distutils.version import LooseVersion\n",
      "      Processing numpy/random\\mtrand.pyx\n",
      "      Processing numpy/random\\_bounded_integers.pyx.in\n",
      "      Processing numpy/random\\_common.pyx\n",
      "      Processing numpy/random\\_generator.pyx\n",
      "      Processing numpy/random\\_mt19937.pyx\n",
      "      Processing numpy/random\\_pcg64.pyx\n",
      "      Processing numpy/random\\_philox.pyx\n",
      "      Processing numpy/random\\_sfc64.pyx\n",
      "      INFO: blas_opt_info:\n",
      "      INFO: blas_armpl_info:\n",
      "      INFO: No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils\n",
      "      INFO: customize MSVCCompiler\n",
      "      INFO:   libraries armpl_lp64_mp not found in ['C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\lib', 'C:\\\\', 'C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\libs']\n",
      "      INFO:   NOT AVAILABLE\n",
      "      INFO:\n",
      "      INFO: blas_mkl_info:\n",
      "      INFO:   libraries mkl_rt not found in ['C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\lib', 'C:\\\\', 'C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\libs']\n",
      "      INFO:   NOT AVAILABLE\n",
      "      INFO:\n",
      "      INFO: blis_info:\n",
      "      INFO:   libraries blis not found in ['C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\lib', 'C:\\\\', 'C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\libs']\n",
      "      INFO:   NOT AVAILABLE\n",
      "      INFO:\n",
      "      INFO: openblas_info:\n",
      "      INFO:   libraries openblas not found in ['C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\lib', 'C:\\\\', 'C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\libs']\n",
      "      INFO: get_default_fcompiler: matching types: '['gnu', 'intelv', 'absoft', 'compaqv', 'intelev', 'gnu95', 'g95', 'intelvem', 'intelem', 'flang']'\n",
      "      INFO: customize GnuFCompiler\n",
      "      WARN: Could not locate executable g77\n",
      "      WARN: Could not locate executable f77\n",
      "      INFO: customize IntelVisualFCompiler\n",
      "      WARN: Could not locate executable ifort\n",
      "      WARN: Could not locate executable ifl\n",
      "      INFO: customize AbsoftFCompiler\n",
      "      WARN: Could not locate executable f90\n",
      "      INFO: customize CompaqVisualFCompiler\n",
      "      WARN: Could not locate executable DF\n",
      "      INFO: customize IntelItaniumVisualFCompiler\n",
      "      WARN: Could not locate executable efl\n",
      "      INFO: customize Gnu95FCompiler\n",
      "      WARN: Could not locate executable gfortran\n",
      "      WARN: Could not locate executable f95\n",
      "      INFO: customize G95FCompiler\n",
      "      WARN: Could not locate executable g95\n",
      "      INFO: customize IntelEM64VisualFCompiler\n",
      "      INFO: customize IntelEM64TFCompiler\n",
      "      WARN: Could not locate executable efort\n",
      "      WARN: Could not locate executable efc\n",
      "      INFO: customize PGroupFlangCompiler\n",
      "      WARN: Could not locate executable flang\n",
      "      WARN: don't know how to compile Fortran code on platform 'nt'\n",
      "      INFO:   NOT AVAILABLE\n",
      "      INFO:\n",
      "      INFO: accelerate_info:\n",
      "      INFO:   NOT AVAILABLE\n",
      "      INFO:\n",
      "      INFO: atlas_3_10_blas_threads_info:\n",
      "      INFO: Setting PTATLAS=ATLAS\n",
      "      INFO:   libraries tatlas not found in ['C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\lib', 'C:\\\\', 'C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\libs']\n",
      "      INFO:   NOT AVAILABLE\n",
      "      INFO:\n",
      "      INFO: atlas_3_10_blas_info:\n",
      "      INFO:   libraries satlas not found in ['C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\lib', 'C:\\\\', 'C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\libs']\n",
      "      INFO:   NOT AVAILABLE\n",
      "      INFO:\n",
      "      INFO: atlas_blas_threads_info:\n",
      "      INFO: Setting PTATLAS=ATLAS\n",
      "      INFO:   libraries ptf77blas,ptcblas,atlas not found in ['C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\lib', 'C:\\\\', 'C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\libs']\n",
      "      INFO:   NOT AVAILABLE\n",
      "      INFO:\n",
      "      INFO: atlas_blas_info:\n",
      "      INFO:   libraries f77blas,cblas,atlas not found in ['C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\lib', 'C:\\\\', 'C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\libs']\n",
      "      INFO:   NOT AVAILABLE\n",
      "      INFO:\n",
      "      C:\\Users\\Eleaquim\\AppData\\Local\\Temp\\pip-install-_4vnv0i1\\numpy_267767f23bd142af96a61a29b987ffbb\\numpy\\distutils\\system_info.py:2077: UserWarning:\n",
      "          Optimized (vendor) Blas libraries are not found.\n",
      "          Falls back to netlib Blas library which has worse performance.\n",
      "          A better performance should be easily gained by switching\n",
      "          Blas library.\n",
      "        if self._calc_info(blas):\n",
      "      INFO: blas_info:\n",
      "      INFO:   libraries blas not found in ['C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\lib', 'C:\\\\', 'C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\libs']\n",
      "      INFO:   NOT AVAILABLE\n",
      "      INFO:\n",
      "      C:\\Users\\Eleaquim\\AppData\\Local\\Temp\\pip-install-_4vnv0i1\\numpy_267767f23bd142af96a61a29b987ffbb\\numpy\\distutils\\system_info.py:2077: UserWarning:\n",
      "          Blas (http://www.netlib.org/blas/) libraries not found.\n",
      "          Directories to search for the libraries can be specified in the\n",
      "          numpy/distutils/site.cfg file (section [blas]) or by setting\n",
      "          the BLAS environment variable.\n",
      "        if self._calc_info(blas):\n",
      "      INFO: blas_src_info:\n",
      "      INFO:   NOT AVAILABLE\n",
      "      INFO:\n",
      "      C:\\Users\\Eleaquim\\AppData\\Local\\Temp\\pip-install-_4vnv0i1\\numpy_267767f23bd142af96a61a29b987ffbb\\numpy\\distutils\\system_info.py:2077: UserWarning:\n",
      "          Blas (http://www.netlib.org/blas/) sources not found.\n",
      "          Directories to search for the sources can be specified in the\n",
      "          numpy/distutils/site.cfg file (section [blas_src]) or by setting\n",
      "          the BLAS_SRC environment variable.\n",
      "        if self._calc_info(blas):\n",
      "      INFO:   NOT AVAILABLE\n",
      "      INFO:\n",
      "      non-existing path in 'numpy\\\\distutils': 'site.cfg'\n",
      "      INFO: lapack_opt_info:\n",
      "      INFO: lapack_armpl_info:\n",
      "      INFO:   libraries armpl_lp64_mp not found in ['C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\lib', 'C:\\\\', 'C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\libs']\n",
      "      INFO:   NOT AVAILABLE\n",
      "      INFO:\n",
      "      INFO: lapack_mkl_info:\n",
      "      INFO:   libraries mkl_rt not found in ['C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\lib', 'C:\\\\', 'C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\libs']\n",
      "      INFO:   NOT AVAILABLE\n",
      "      INFO:\n",
      "      INFO: openblas_lapack_info:\n",
      "      INFO:   libraries openblas not found in ['C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\lib', 'C:\\\\', 'C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\libs']\n",
      "      INFO:   NOT AVAILABLE\n",
      "      INFO:\n",
      "      INFO: openblas_clapack_info:\n",
      "      INFO:   libraries openblas,lapack not found in ['C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\lib', 'C:\\\\', 'C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\libs']\n",
      "      INFO:   NOT AVAILABLE\n",
      "      INFO:\n",
      "      INFO: flame_info:\n",
      "      INFO:   libraries flame not found in ['C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\lib', 'C:\\\\', 'C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\libs']\n",
      "      INFO:   NOT AVAILABLE\n",
      "      INFO:\n",
      "      INFO: atlas_3_10_threads_info:\n",
      "      INFO: Setting PTATLAS=ATLAS\n",
      "      INFO:   libraries tatlas,tatlas not found in C:\\Users\\Eleaquim\\AppData\\Local\\Programs\\Python\\Python310-32\\lib\n",
      "      INFO:   libraries tatlas,tatlas not found in C:\\\n",
      "      INFO:   libraries tatlas,tatlas not found in C:\\Users\\Eleaquim\\AppData\\Local\\Programs\\Python\\Python310-32\\libs\n",
      "      INFO: <class 'numpy.distutils.system_info.atlas_3_10_threads_info'>\n",
      "      INFO:   NOT AVAILABLE\n",
      "      INFO:\n",
      "      INFO: atlas_3_10_info:\n",
      "      INFO:   libraries satlas,satlas not found in C:\\Users\\Eleaquim\\AppData\\Local\\Programs\\Python\\Python310-32\\lib\n",
      "      INFO:   libraries satlas,satlas not found in C:\\\n",
      "      INFO:   libraries satlas,satlas not found in C:\\Users\\Eleaquim\\AppData\\Local\\Programs\\Python\\Python310-32\\libs\n",
      "      INFO: <class 'numpy.distutils.system_info.atlas_3_10_info'>\n",
      "      INFO:   NOT AVAILABLE\n",
      "      INFO:\n",
      "      INFO: atlas_threads_info:\n",
      "      INFO: Setting PTATLAS=ATLAS\n",
      "      INFO:   libraries ptf77blas,ptcblas,atlas not found in C:\\Users\\Eleaquim\\AppData\\Local\\Programs\\Python\\Python310-32\\lib\n",
      "      INFO:   libraries ptf77blas,ptcblas,atlas not found in C:\\\n",
      "      INFO:   libraries ptf77blas,ptcblas,atlas not found in C:\\Users\\Eleaquim\\AppData\\Local\\Programs\\Python\\Python310-32\\libs\n",
      "      INFO: <class 'numpy.distutils.system_info.atlas_threads_info'>\n",
      "      INFO:   NOT AVAILABLE\n",
      "      INFO:\n",
      "      INFO: atlas_info:\n",
      "      INFO:   libraries f77blas,cblas,atlas not found in C:\\Users\\Eleaquim\\AppData\\Local\\Programs\\Python\\Python310-32\\lib\n",
      "      INFO:   libraries f77blas,cblas,atlas not found in C:\\\n",
      "      INFO:   libraries f77blas,cblas,atlas not found in C:\\Users\\Eleaquim\\AppData\\Local\\Programs\\Python\\Python310-32\\libs\n",
      "      INFO: <class 'numpy.distutils.system_info.atlas_info'>\n",
      "      INFO:   NOT AVAILABLE\n",
      "      INFO:\n",
      "      INFO: lapack_info:\n",
      "      INFO:   libraries lapack not found in ['C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\lib', 'C:\\\\', 'C:\\\\Users\\\\Eleaquim\\\\AppData\\\\Local\\\\Programs\\\\Python\\\\Python310-32\\\\libs']\n",
      "      INFO:   NOT AVAILABLE\n",
      "      INFO:\n",
      "      C:\\Users\\Eleaquim\\AppData\\Local\\Temp\\pip-install-_4vnv0i1\\numpy_267767f23bd142af96a61a29b987ffbb\\numpy\\distutils\\system_info.py:1902: UserWarning:\n",
      "          Lapack (http://www.netlib.org/lapack/) libraries not found.\n",
      "          Directories to search for the libraries can be specified in the\n",
      "          numpy/distutils/site.cfg file (section [lapack]) or by setting\n",
      "          the LAPACK environment variable.\n",
      "        return getattr(self, '_calc_info_{}'.format(name))()\n",
      "      INFO: lapack_src_info:\n",
      "      INFO:   NOT AVAILABLE\n",
      "      INFO:\n",
      "      C:\\Users\\Eleaquim\\AppData\\Local\\Temp\\pip-install-_4vnv0i1\\numpy_267767f23bd142af96a61a29b987ffbb\\numpy\\distutils\\system_info.py:1902: UserWarning:\n",
      "          Lapack (http://www.netlib.org/lapack/) sources not found.\n",
      "          Directories to search for the sources can be specified in the\n",
      "          numpy/distutils/site.cfg file (section [lapack_src]) or by setting\n",
      "          the LAPACK_SRC environment variable.\n",
      "        return getattr(self, '_calc_info_{}'.format(name))()\n",
      "      INFO:   NOT AVAILABLE\n",
      "      INFO:\n",
      "      INFO: numpy_linalg_lapack_lite:\n",
      "      INFO:   FOUND:\n",
      "      INFO:     language = c\n",
      "      INFO:\n",
      "      Warning: attempted relative import with no known parent package\n",
      "      C:\\Users\\Eleaquim\\AppData\\Local\\Programs\\Python\\Python310-32\\lib\\distutils\\dist.py:274: UserWarning: Unknown distribution option: 'define_macros'\n",
      "        warnings.warn(msg)\n",
      "      running bdist_wheel\n",
      "      running build\n",
      "      running config_cc\n",
      "      INFO: unifing config_cc, config, build_clib, build_ext, build commands --compiler options\n",
      "      running config_fc\n",
      "      INFO: unifing config_fc, config, build_clib, build_ext, build commands --fcompiler options\n",
      "      running build_src\n",
      "      INFO: build_src\n",
      "      INFO: building py_modules sources\n",
      "      creating build\n",
      "      creating build\\src.win32-3.10\n",
      "      creating build\\src.win32-3.10\\numpy\n",
      "      creating build\\src.win32-3.10\\numpy\\distutils\n",
      "      INFO: building library \"npymath\" sources\n",
      "      error: Microsoft Visual C++ 14.0 or greater is required. Get it with \"Microsoft C++ Build Tools\": https://visualstudio.microsoft.com/visual-cpp-build-tools/\n",
      "      [end of output]\n",
      "  \n",
      "  note: This error originates from a subprocess, and is likely not a problem with pip.\n",
      "  ERROR: Failed building wheel for numpy\n",
      "ERROR: Could not build wheels for numpy, which is required to install pyproject.toml-based projects\n",
      "WARNING: Ignoring invalid distribution -ip (c:\\users\\eleaquim\\appdata\\local\\programs\\python\\python310-32\\lib\\site-packages)\n",
      "WARNING: Ignoring invalid distribution -ip (c:\\users\\eleaquim\\appdata\\local\\programs\\python\\python310-32\\lib\\site-packages)\n"
     ]
    }
   ],
   "source": [
    " %pip install matplotlib\n",
    " %pip install numpy"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Niveles de Oxígeno\n",
    "\n",
    "Muestra diez minutos de niveles de oxígeno en tu nave."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "ename": "IndentationError",
     "evalue": "unexpected indent (1036842675.py, line 2)",
     "output_type": "error",
     "traceback": [
      "\u001b[1;36m  Input \u001b[1;32mIn [34]\u001b[1;36m\u001b[0m\n\u001b[1;33m    import matplotlib.pyplot as plt\u001b[0m\n\u001b[1;37m    ^\u001b[0m\n\u001b[1;31mIndentationError\u001b[0m\u001b[1;31m:\u001b[0m unexpected indent\n"
     ]
    }
   ],
   "source": [
    "   import numpy as np\n",
    "    import matplotlib.pyplot as plt\n",
    "    data = np.random.default_rng(12345)\n",
    "    oxy_nums = data.integers(low=0, high=10, size=10)\n",
    "\n",
    "    plt.bar(range(len(oxy_nums)), oxy_nums)\n",
    "    plt.show()\n",
    "    "
   ]
  }
 ],
 "metadata": {
  "interpreter": {
   "hash": "89a16dd1826562c6c3c806b0f1d3f41b250df0a4bbadc199da1b00b9b1e3eacc"
  },
  "kernelspec": {
   "display_name": "Python 3.10.2 32-bit",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.2"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
